### PR TITLE
Providing support to enable tracing in Executors viz. java.unit.concurrent package

### PR DESCRIPTION
**Pull request by gauravrmazra**
I have added classes to provide support for tracing in executors. I was evaluating to use spring-cloud-sleuth in one of our project for tracing. I saw it was not providing support for Executors. Now, I have written a glue code to provide support, this is what I wrote to support spring-cloud-sleuth in our application. I hope this will be helpful to this project as well.

### CODE REVIEW

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/concurrent/executor/support/TraceCallableSupport.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * A helper class to decorate {@link Callable} to {@link TraceCallable}
+ * @author Gaurav Rai Mazra
+ *
+ */
+public class TraceCallableSupport {
+	private TraceCallableSupport() {
+		
+	}
+	
+	public static <V> Callable<V> decorate(Trace trace, Callable<V> command) {
```
Comment (spencergibb) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/concurrent/executor/support/TraceCallableSupport.java]: Seems redundant to me. Just replace it with `new TraceCallable<>(trace, command);`

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/concurrent/executor/support/TraceRunnableSupport.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * A helper class to decorate {@link Runnable} to {@link TraceRunnable}
+ * @author Gaurav Rai Mazra
+ *
+ */
+public class TraceRunnableSupport {
+	private TraceRunnableSupport() {
+		
+	}
+	
+	public static Runnable decorate(Trace trace, Runnable delegate) {
```
Comment (spencergibb) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/concurrent/executor/support/TraceRunnableSupport.java]: Seems redundant to me. Just replace it with new TraceRunnable(trace, delegate);

**Comment by dsyer** [comment]
Maybe a test or two would smooth this one on its way?

**Comment by gauravrmazra** [comment]
I will sure write test cases for this.

**Comment by gauravrmazra** [comment]
I have written a test case for TraceableExecutorService and to support my claim in issue
https://github.com/spring-cloud/spring-cloud-sleuth/issues/60

**Pull request closed without merging**